src/gateway.rs

```diff
@@ -64,29 +65,36 @@ impl MessageSender {
 
 pub struct Gateway {
     uplinks: dispatcher::MessageSender,
+    self_sender: MessageSender,
     messages: MessageReceiver,
     beacon_handler: beaconer::MessageSender,
     downlink_mac: MacAddress,
     udp_runtime: UdpRuntime,
     listen_address: String,
     region_params: Option<RegionParams>,
+    /// If packet forwarder returns InvalidTxPower, it provides the max allowable tx value
+    /// which we can save and apply it as a maximum to outgoing packets
```

Comment:
so the invalid tx value is returned but the packet is still sent with that indicated tx power  right?

Reply:
The InvalidTxValue originally stems from the packet forwarder which will NACK a packet with power that is too high; the frame is dropped by the packet forwarder.

In that NACK, however, it provides the max allowable power. After that, gateway-rs will never trigger the error again from packet forwarder as it saves the max_power value and applies to to future packets; it does this quietly within `fn tx_power` without alerting the user.

Reply:
oh man it's dropped by the packet forwarder? I was convinced it would return what power it transmitted at